### DISCUSSION TITLE
Is it possible to change the ID for an item programatically when creating it?

### DISCUSSION THREAD

**Discussion by ametis70** [payloadcms/payload#9814] [Q&A]
I'm trying to make a script that seeds a few collections. So far I was able to make it work for collections that don't have relationships with other collections, but when I tried to reference some upload that belongs to another collection, I realized that the ID saved on the seed was different to the one in the upload when I ran the script to recreate all the data on the db.

Naturally, I tried passing the ID to the `data` object when creating the documents, but payload ignored it. After that, I  tried enabling custom IDs in the collection but it still didn't work. I also tried running `payload.update` right after `payload.create` but that didn't work either. My last experiment was with an `afterOperation` hook, but it also failed.

For the record I'm using MongoDB, and I was able to do this on the past (using Payload 1.8.3). Maybe after so many versions it's no longer possible? I found https://github.com/payloadcms/payload/issues/6884 but I thought that maybe it's an issue exclusive to Postgres.

Thanks!

**Comment by DanRibbens** [comment]
Passing id in your data object on create should be all you need to do. Can you share your id field config and your code calling create?

**Comment by ametis70** [comment]
As I couldn't make it work, I never made a commit with that code and ended up using [mongo-seeding](https://github.com/pkosiec/mongo-seeding). I'm kinda rushing something right now, but I'll make a repo to try to reproduce it in a few days, because I'm interested in making a plugin for Payload that does this in a db agnostic way
